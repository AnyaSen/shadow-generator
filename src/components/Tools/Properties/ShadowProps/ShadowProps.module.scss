@import "../../../../assets/scss/variables";

.ShadowProps {
  margin: $space-2;
  h1 {
    margin-bottom: $space-2;
  }
}
