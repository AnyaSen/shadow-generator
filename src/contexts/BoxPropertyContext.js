import React, { useState, createContext } from "react";

export const BoxPropertyContext = createContext();

export const BoxPropertyProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(350);
  const [widthValue, setWidthValue] = useState(350);
  const [radiusValue, setRadiusValue] = useState(16);
  const [borderValue, setBorderValue] = useState(0);

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };

  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };

  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };

  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };

  return (
    <BoxPropertyContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue
      }}
    >
      {children}
    </BoxPropertyContext.Provider>
  );
};
