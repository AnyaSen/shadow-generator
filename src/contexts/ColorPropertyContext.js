import React, { useState, createContext } from "react";

export const ColorPropertyContext = createContext();

export const ColorPropertyProvider = ({ children }) => {
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ColorPropertyContext.Provider
      value={{
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ColorPropertyContext.Provider>
  );
};
