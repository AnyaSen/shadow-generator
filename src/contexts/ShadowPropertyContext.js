import React, { useState, createContext } from "react";

export const ShadowPropertyContext = createContext();

export const ShadowPropertyProvider = ({ children }) => {
  const [shiftHorizontically, setShiftHorizontically] = useState(20);
  const [shiftVertically, setShiftVertically] = useState(14);
  const [blurValue, setBlurValue] = useState(10);
  const [opacityValue, setOpacityValue] = useState(12);

  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };

  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };

  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };

  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };

  return (
    <ShadowPropertyContext.Provider
      value={{
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue
      }}
    >
      {children}
    </ShadowPropertyContext.Provider>
  );
};
