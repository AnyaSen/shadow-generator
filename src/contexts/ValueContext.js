import React, { useState, createContext } from "react";

export const ValueContext = createContext();

export const ValueProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(45);
  const [widthValue, setWidthValue] = useState(45);
  const [radiusValue, setRadiusValue] = useState(1);
  const [shiftHorizontically, setShiftHorizontically] = useState(1.25);
  const [shiftVertically, setShiftVertically] = useState(0.85);
  const [blurValue, setBlurValue] = useState(0.625);
  const [opacityValue, setOpacityValue] = useState(0.75);
  const [borderValue, setBorderValue] = useState(0);
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };
  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };
  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };
  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };
  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };
  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };
  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };
  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };
  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ValueContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue,
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue,
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ValueContext.Provider>
  );
};
