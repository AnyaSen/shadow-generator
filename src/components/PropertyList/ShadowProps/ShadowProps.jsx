import React, { useContext } from "react";
import Styles from "./ShadowProps.module.scss";

import ValueSlider from "../../ValueSlider/ValueSlider";

import { ShadowPropertyContext } from "../../../contexts/ShadowPropertyContext";

export default function ShadowProps() {
  const {
    shiftHorizontically,
    ChangeShiftHorizontically,
    shiftVertically,
    ChangeShiftVertically,
    blurValue,
    ChangeBlurValue,
    opacityValue,
    ChangeOpacityValue
  } = useContext(ShadowPropertyContext);

  return (
    <div className={Styles.ShadowProps} data-testid="shadowProps">
      <h1>SHADOW</h1>

      <ValueSlider
        heading="shift horizontically"
        value={shiftHorizontically}
        handleChange={ChangeShiftHorizontically}
        maxValue={200}
        minValue={-200}
      />

      <ValueSlider
        heading="shift vertically"
        value={shiftVertically}
        handleChange={ChangeShiftVertically}
        maxValue={100}
        minValue={-100}
      />

      <ValueSlider
        heading="blur"
        value={blurValue}
        handleChange={ChangeBlurValue}
        maxValue={100}
      />

      <ValueSlider
        heading="opacity"
        value={opacityValue}
        handleChange={ChangeOpacityValue}
        maxValue={1}
        minValue={0}
      />
    </div>
  );
}
