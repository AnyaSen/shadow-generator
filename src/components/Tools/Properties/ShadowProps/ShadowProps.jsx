import React, { useContext } from "react";
import Styles from "./ShadowProps.module.scss";
import { ValueContext } from "../../../../contexts/ValueContext";

import ValueSlider from "../../../ValueSlider/ValueSlider";

export default function ShadowProps() {
  const {
    shiftHorizontically,
    ChangeShiftHorizontically,
    shiftVertically,
    ChangeShiftVertically,
    blurValue,
    ChangeBlurValue,
    opacityValue,
    ChangeOpacityValue
  } = useContext(ValueContext);

  return (
    <div className={Styles.ShadowProps}>
      <h1>SHADOW</h1>
      <ValueSlider
        heading="shift horizontically"
        value={shiftHorizontically}
        handleChange={ChangeShiftHorizontically}
        maxValue={15}
        minValue={-15}
      />
      <ValueSlider
        heading="shift vertically"
        value={shiftVertically}
        handleChange={ChangeShiftVertically}
        maxValue={10}
        minValue={-10}
      />
      <ValueSlider
        heading="blur"
        value={blurValue}
        handleChange={ChangeBlurValue}
        maxValue={10}
        minValue={0}
      />
      <ValueSlider
        heading="opacity"
        value={opacityValue}
        handleChange={ChangeOpacityValue}
        maxValue={1}
        minValue={0}
      />
    </div>
  );
}
