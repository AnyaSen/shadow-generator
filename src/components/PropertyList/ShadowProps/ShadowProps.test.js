import React from "react";
import { render, cleanup } from "@testing-library/react";

import ShadowProps from "./ShadowProps";

import { ShadowPropertyProvider } from "../../../contexts/ShadowPropertyContext";

describe("<ShadowProps />", () => {
  afterEach(cleanup);

  const createRenderTree = () => (
    <ShadowPropertyProvider>
      <ShadowProps />
    </ShadowPropertyProvider>
  );

  test("Should return 5 children and correct text.", () => {
    const tree = createRenderTree();
    const text = "SHADOW";
    const { getByTestId, getByText, container } = render(tree);

    const shadowPropsContainer = getByTestId(`shadowProps`);
    const shadowPropsHeading = getByText(text);

    expect(shadowPropsContainer.children.length).toEqual(5);
    expect(shadowPropsHeading).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
