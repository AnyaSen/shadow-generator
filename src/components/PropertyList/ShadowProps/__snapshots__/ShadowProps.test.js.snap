// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShadowProps /> Should return 5 children and correct text. 1`] = `
<div
  class="ShadowProps"
  data-testid="shadowProps"
>
  <h1>
    SHADOW
  </h1>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      shift horizontically
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 55%;"
      />
      <input
        type="hidden"
        value="20"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="200"
        aria-valuemin="-200"
        aria-valuenow="20"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 55%;"
        tabindex="0"
      />
    </span>
  </div>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      shift vertically
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 57%;"
      />
      <input
        type="hidden"
        value="14"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="-100"
        aria-valuenow="14"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 57%;"
        tabindex="0"
      />
    </span>
  </div>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      blur
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 10%;"
      />
      <input
        type="hidden"
        value="10"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 10%;"
        tabindex="0"
      />
    </span>
  </div>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      opacity
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 100%;"
      />
      <input
        type="hidden"
        value="1"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="1"
        aria-valuemin="0"
        aria-valuenow="1"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 100%;"
        tabindex="0"
      />
    </span>
  </div>
</div>
`;
